pkg/util/shell/shell.go

```diff
@@ -17,9 +17,15 @@ limitations under the License.
 package shell
 
 import (
+	"errors"
 	"fmt"
 	"github.com/docker/machine/libmachine/shell"
 	"os"
+	"strings"
+)
+
+var (
+	supportedShell = []string{"bash", "fish", "powershell", "cmd", "emacs", "tcsh", "zsh"}
```

Comment:
+1 we could have gotten the list of shells also from shells.go, but fair enough.